File: Code/Quickstarts/vb/UIProcessQuickstarts_AdvancedHostDemo/Client/PhoneControl.vb
 '===============================================================================
' Microsoft User Interface Process Application Block for .NET Quick Start
' http://msdn.microsoft.com/library/en-us/dnbda/html/uip.asp
'
' PhoneControl.vb
'
' This file contains the implementations of the PhoneControl class.
'
' 
'===============================================================================
' Copyright (C) 2000-2001 Microsoft Corporation
' All rights reserved.
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
' OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
' LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR
' FITNESS FOR A PARTICULAR PURPOSE.
'==============================================================================

Imports System
Imports System.Collections
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Windows.Forms

Namespace Client

    ' <summary>
    ' Summary description for PhoneControl.
    ' </summary>
    Public Class PhoneControl
        Inherits System.Windows.Forms.UserControl
        Private label1 As System.Windows.Forms.Label
        Private label2 As System.Windows.Forms.Label
        Private areaCodeTxt As System.Windows.Forms.TextBox
        Private phoneNumTxt As System.Windows.Forms.TextBox

        Private phone As EmployeeData.PhoneRow

        ' <summary> 
        ' Required designer variable.
        ' </summary>
        Private components As System.ComponentModel.Container = Nothing

        Public Sub New()
            ' This call is required by the Windows.Forms Form Designer.
            InitializeComponent()
            ' TODO: Add any initialization after the InitializeComponent call
        End Sub

        Public Sub RefreshData()
            If Not (phone Is Nothing) Then
                areaCodeTxt.DataBindings.Clear()
                areaCodeTxt.DataBindings.Add("Text", phone, "AreaCode")
                phoneNumTxt.DataBindings.Clear()
                phoneNumTxt.DataBindings.Add("Text", phone, "PhoneNumber")
            End If
        End Sub

        Public WriteOnly Property PhoneNumber() As EmployeeData.PhoneRow
            Set(ByVal value As EmployeeData.PhoneRow)
                phone = value
                If Not (phone Is Nothing) Then
                    RefreshData()
                End If
            End Set
        End Property

        ' <summary> 
        ' Clean up any resources being used.
        ' </summary>
        Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
            If disposing Then
                If Not (components Is Nothing) Then
                    components.Dispose()
                End If
            End If
            MyBase.Dispose(disposing)
        End Sub

#Region "Component Designer generated code"

        ' <summary> 
        ' Required method for Designer support - do not modify 
        ' the contents of this method with the code editor.
        ' </summary>
        Private Sub InitializeComponent()
            Me.label1 = New System.Windows.Forms.Label
            Me.label2 = New System.Windows.Forms.Label
            Me.areaCodeTxt = New System.Windows.Forms.TextBox
            Me.phoneNumTxt = New System.Windows.Forms.TextBox
            Me.SuspendLayout()
            ' 
            ' label1
            ' 
            Me.label1.Location = New System.Drawing.Point(24, 24)
            Me.label1.Name = "label1"
            Me.label1.Size = New System.Drawing.Size(64, 16)
            Me.label1.TabIndex = 0
            Me.label1.Text = "Area Code"
            ' 
            ' label2
            ' 
            Me.label2.Location = New System.Drawing.Point(24, 64)
            Me.label2.Name = "label2"
            Me.label2.Size = New System.Drawing.Size(48, 16)
            Me.label2.TabIndex = 1
            Me.label2.Text = "Phone #"
            ' 
            ' areaCodeTxt
            ' 
            Me.areaCodeTxt.Location = New System.Drawing.Point(88, 24)
            Me.areaCodeTxt.MaxLength = 3
            Me.areaCodeTxt.Name = "areaCodeTxt"
            Me.areaCodeTxt.Size = New System.Drawing.Size(48, 20)
            Me.areaCodeTxt.TabIndex = 2
            Me.areaCodeTxt.Text = ""
            ' 
            ' phoneNumTxt
            ' 
            Me.phoneNumTxt.Location = New System.Drawing.Point(88, 64)
            Me.phoneNumTxt.MaxLength = 8
            Me.phoneNumTxt.Name = "phoneNumTxt"
            Me.phoneNumTxt.Size = New System.Drawing.Size(88, 20)
            Me.phoneNumTxt.TabIndex = 3
            Me.phoneNumTxt.Text = ""
            ' 
            ' PhoneControl
            ' 
            Me.Controls.Add(phoneNumTxt)
            Me.Controls.Add(areaCodeTxt)
            Me.Controls.Add(label2)
            Me.Controls.Add(label1)
            Me.Name = "PhoneControl"
            Me.Size = New System.Drawing.Size(216, 136)
            Me.ResumeLayout(False)
        End Sub

#End Region

    End Class

End Namespace

File: Code/Quickstarts/cs/UIProcessQuickstarts_AdvancedHostDemo/Client/PhoneControl.cs
//===============================================================================
// Microsoft User Interface Process Application Block for .NET Quick Start
// http://msdn.microsoft.com/library/en-us/dnbda/html/uip.asp
//
// PhoneControl.cs
//
// This file contains the implementations of the PhoneControl class.
//
// 
//===============================================================================
// Copyright (C) 2000-2001 Microsoft Corporation
// All rights reserved.
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
// OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
// LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR
// FITNESS FOR A PARTICULAR PURPOSE.
//==============================================================================

using System;
using System.Collections;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Windows.Forms;

namespace Client
{
	/// <summary>
	/// Summary description for PhoneControl.
	/// </summary>
	public class PhoneControl : System.Windows.Forms.UserControl
	{
		private System.Windows.Forms.Label label1;
		private System.Windows.Forms.Label label2;
		private System.Windows.Forms.TextBox areaCodeTxt;
		private System.Windows.Forms.TextBox phoneNumTxt;

		private EmployeeData.PhoneRow phone;		

		/// <summary> 
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		public PhoneControl()
		{
			// This call is required by the Windows.Forms Form Designer.
			InitializeComponent();
			// TODO: Add any initialization after the InitializeComponent call
		}

		public void RefreshData() 
		{
			if( phone != null) 
			{
				areaCodeTxt.DataBindings.Clear();
				areaCodeTxt.DataBindings.Add("Text", phone, "AreaCode");
				phoneNumTxt.DataBindings.Clear();
				phoneNumTxt.DataBindings.Add("Text", phone, "PhoneNumber");
			}
		}

		public EmployeeData.PhoneRow PhoneNumber 
		{
			set 
			{ 
				phone = value;
				if (phone != null) 
				{
					RefreshData();			
				}
			}
		}

		/// <summary> 
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Component Designer generated code
		/// <summary> 
		/// Required method for Designer support - do not modify 
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			this.label1 = new System.Windows.Forms.Label();
			this.label2 = new System.Windows.Forms.Label();
			this.areaCodeTxt = new System.Windows.Forms.TextBox();
			this.phoneNumTxt = new System.Windows.Forms.TextBox();
			this.SuspendLayout();
			// 
			// label1
			// 
			this.label1.Location = new System.Drawing.Point(24, 24);
			this.label1.Name = "label1";
			this.label1.Size = new System.Drawing.Size(64, 16);
			this.label1.TabIndex = 0;
			this.label1.Text = "Area Code";
			// 
			// label2
			// 
			this.label2.Location = new System.Drawing.Point(24, 64);
			this.label2.Name = "label2";
			this.label2.Size = new System.Drawing.Size(48, 16);
			this.label2.TabIndex = 1;
			this.label2.Text = "Phone #";
			// 
			// areaCodeTxt
			// 
			this.areaCodeTxt.Location = new System.Drawing.Point(88, 24);
			this.areaCodeTxt.MaxLength = 3;
			this.areaCodeTxt.Name = "areaCodeTxt";
			this.areaCodeTxt.Size = new System.Drawing.Size(48, 20);
			this.areaCodeTxt.TabIndex = 2;
			this.areaCodeTxt.Text = "";
			// 
			// phoneNumTxt
			// 
			this.phoneNumTxt.Location = new System.Drawing.Point(88, 64);
			this.phoneNumTxt.MaxLength = 8;
			this.phoneNumTxt.Name = "phoneNumTxt";
			this.phoneNumTxt.Size = new System.Drawing.Size(88, 20);
			this.phoneNumTxt.TabIndex = 3;
			this.phoneNumTxt.Text = "";
			// 
			// PhoneControl
			// 
			this.Controls.Add(this.phoneNumTxt);
			this.Controls.Add(this.areaCodeTxt);
			this.Controls.Add(this.label2);
			this.Controls.Add(this.label1);
			this.Name = "PhoneControl";
			this.Size = new System.Drawing.Size(216, 136);
			this.ResumeLayout(false);

		}
		#endregion
	}
}
